concordium-consensus/src/Concordium/ImportExport.hs

```diff
@@ -116,30 +125,122 @@ sectionHeaderLength :: Word64
 sectionHeaderLength = fromIntegral $ BS.length $ encode placeholderSectionHeader
 
 -- |Open a file handle for writing a chunk. If the file with the specified name already exists, a
--- new name is chosen by putting or incrementing a version number in its extension. It is expected
--- that an unversioned filename only has a single extension.
-initialHandle :: FilePath -> IO (FilePath, Handle)
-initialHandle p = do
+-- boolean flag indicates whether the file should be overwritten or whether a new name should be
+-- chosen by adding to or incrementing a version number in its extension. It is expected that an
+-- unversioned filename only has a single extension.
+initialHandle :: Bool -> FilePath -> IO (FilePath, Handle)
+initialHandle overwriteExistingChunk p = do
     createDirectoryIfMissing True (takeDirectory p)
     chunkExists <- doesFileExist p
-    if chunkExists
-        then initialHandle (replaceExtensions p newExt)
+    if chunkExists && not overwriteExistingChunk
```

Comment:
Why make this more complicated than it needs to be with an additional option?

Reply:
I do not see that in the issue description. What I do see is that there is a **suggestion** that the old chunk is deleted in the last item. That suggestion is a good one, and should be implemented. But for simplicity it should just be done, we should not have options unless we need them.

This is also very different than overwriting an existing file. It is essential that we do not use the same names because these files are cached.